stan/math/prim/core/init_threadpool_tbb.hpp

```diff
@@ -5,6 +5,14 @@
 
 #include <boost/lexical_cast.hpp>
 
+#include <tbb/tbb_stddef.h>
```

Comment:
This breaks external oneTBB since `tbb/tbb_stddef.h` is not found:
```bash
stan/math/prim/core/init_threadpool_tbb.hpp(8):
  catastrophic error: cannot open source file "tbb/tbb_stddef.h"
  #include <tbb/tbb_stddef.h>
```


`TBB_VERSION_MAJOR` is now defined in `tbb/version.h`.

Reply:
Thanks! So seems we need to add another ONE_TBB compile time define.

Reply:
I'll submit a hot fix shortly; testing it now.

Reply:
@rok-cesnovar #2452 does the trick.

Reply:
I'll put in a fix pr. I think we can just move `#ifndef TBB_INTERFACE_NEW` up since if they have that defined then we know already we are using the new interface

Reply:
Yeah, that is what @hsbadr did in #2452

Reply:
Exactly :)